### ISSUE TITLE
replicating your results of BankHeist

### ISSUE THREAD

**Issue by yueyang130** [mila-iqia/spr#32]
Hi, thanks for yuor amazing work!

I successfully replicated your results **except** BankHeist.  I run BankHeist over 15 seeds and get following results.

command I use is ` python -m scripts.run --game bank_heist --seed ${seed}  --momentum-tau 1. `

15.6 66.1 10.7 23 108.8 61.4 69.7 36.1 22.9 120 29.2 49.4 67.6 29.7 58.4
mean is 51, std is 32

The results I get is far from the results reported in your paper,  380.9 over 10 seeds.

Additionally, there are 100 results of BankHeist from paper [Deep Reinforcement Learning at the Edge of the Statistical Precipice](https://arxiv.org/abs/2108.13264) as follows.

207.4, 492.3, 234.9, 189.3, 267.8, 578.8, 258.4, 861.9, 143.9,
       574.2, 558.7, 282.1, 559.3, 526.7, 232.2, 171. , 242.5, 208.4,
       545.9, 631.2, 140.8, 649.1, 754.4,  43.1, 309.1, 237.6,  67.6,
       194.9, 276.9, 346.2, 594. , 184.5, 592.6, 607. , 575. , 174.2,
       272.9, 149.9, 589.7, 167.2, 185.7, 286.3, 659.2, 259.3, 153.9,
       116.9, 158. , 572.3, 555.8, 687. ,  69.6, 229.5, 206. , 186.1,
       202.3, 225.9, 621.9, 786.3, 727.7, 223.8, 216. , 112.7, 416. ,
       182.1, 269.6, 188.7,  95.1, 506.5, 588.2, 309.5, 505. , 145.1,
       202.2, 195.6, 605.7, 166.4, 810.1, 562.6, 548.1, 193.5, 188.5,
       174.4, 256.4, 230.5, 135.5, 197.3, 297.4, 636.5, 482.6, 241.8,
       223.9, 129.9, 188.4, 175.3, 200.6, 160. , 304.4, 795.5, 191.2,
       708.2
mean is 345, std is 210

There is obvious gap between two distributions. Most of my runnings is under 70 scores while most of the 100 runnings is above 200.

Do you have any idea about why? Or there is a special hyperparameter setting or wrapper setting for game bank_heist?

**Comment by MaxASchwarzer** [comment]
Honestly, not sure. The results from Precipice were generated with this codebase (running on Google servers), so I think they're pretty authoritative. I don't have any experience with Bank Heist itself to be able to say what might be going on there. I guess it's possible that something has changed in the random initialization or the environment itself since then, but it's hard to know without a lot more digging.

**Comment by yueyang130** [comment]
Thanks for your reply. Can you tell me the version of gym and atari_py you used?

Additionally ,except the two repo above, do you think there is any other repo's version difference may cause the problem?

**Comment by MaxASchwarzer** [comment]
God, I don't think I have that conda env anymore - it was just whatever was current in 2020. It's also possible that a PyTorch version could cause a problem, if it changed inits. But really, performance on individual games can be _extremely_ unstable and I wouldn't worry about it too much; even my own Jax reimplementation of SPR has different performance on a lot of individual games. As long as the aggregates are roughly consistent that's about the most you can hope for in Atari.
